irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerService.java

```diff
@@ -44,20 +44,34 @@ public class PolicyCheckerService {
     private final ConstraintCheckerService constraintCheckerService;
 
     public boolean isValid(final Policy policy, final String bpn) {
-        return policy.getPermissions().stream().allMatch(permission -> isValid(permission, getValidStoredPolicies(bpn)));
+        final List<AcceptedPolicy> validStoredPolicies = getValidStoredPolicies(bpn);
+        return policy.getPermissions()
+                     .stream()
+                     .allMatch(permission -> hasValidConstraints(permission, validStoredPolicies));
     }
 
-    private boolean isValid(final Permission permission, final List<AcceptedPolicy> validStoredPolicies) {
+    private boolean hasValidConstraints(final Permission permission, final List<AcceptedPolicy> validStoredPolicies) {
```

Comment:
Introduction of  a new term "constraint" - Is permission not the right term to use here? What are constraints in this relation? What is checked here specifically the permission term a constraint inside the permission or the policy itself?

Reply:
This is not a new term. Permissions consists out of constraints. We have to check all constraints of a permission against the configured ones in order to determine a valid policy as a whole.